opencolorio_config_aces/clf/transforms/arri/generate.py

```diff
@@ -0,0 +1,445 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright Contributors to the OpenColorIO Project.
+"""
+*ARRI* CLF Transforms Generation
+=======================================
+
+Defines procedures for generating ARRI *Common LUT Format* (CLF)
+transforms:
+
+-   :func:`opencolorio_config_aces.clf.generate_clf_arri`
+"""
+
+from math import log, log10
+import PyOpenColorIO as ocio
+from pathlib import Path
+import sys
+
+from opencolorio_config_aces.clf.transforms import (
+    clf_basename,
+    format_clf_transform_id,
+    generate_clf_transform,
+    matrix_RGB_to_RGB_transform,
+)
+from opencolorio_config_aces.config import transform_factory
+
+__author__ = "OpenColorIO Contributors"
+__copyright__ = "Copyright Contributors to the OpenColorIO Project."
+__license__ = "New BSD License - https://opensource.org/licenses/BSD-3-Clause"
+__maintainer__ = "OpenColorIO Contributors"
+__email__ = "ocio-dev@lists.aswf.io"
+__status__ = "Production"
+
+__all__ = [
+    "FAMILY",
+    "GENUS",
+    "VERSION",
+    "generate_clf_arri",
+]
+
+FAMILY = "ARRI"
+"""
+*CLF* transforms family.
+"""
+
+GENUS = "Input"
+"""
+*CLF* transforms genus.
+"""
+
+VERSION = "1.0"
+"""
+*CLF* transforms version.
+"""
+
+
+def _build_awg3_mtx():
+    """
+    Build the `MatrixTransform` for ARRI Wide Gamut 3 primaries.
+
+    Matrix precision is insufficient for Float64 comparison in the LogC
+    White Paper, values are derived from the primaries here.
+
+    Returns
+    -------
+    ocio.MatrixTransform
+         *OpenColorIO* `MatrixTransform`.
+    """
+
+    mtx = matrix_RGB_to_RGB_transform(
+        "ALEXA Wide Gamut", "ACES2065-1", "CAT02"
+    )
+    return mtx
+
+
+def _build_logc3_curve(ei=800, info=False):
+    """
+    Build the `LogCameraTransform` for ARRI LogC3 Curve.
+
+    Parameter values are derived from the published aces-dev IDT formula.
+
+    Parameters
+    ----------
+    ei : int, optional
+        *Exposure Index* of the LogC3 Curve to generate.
+    info : bool, optional
+        Whether to print additional informative text.
+
+    Returns
+    -------
+    ocio.LogCameraTransform
+         *OpenColorIO* `LogCameraTransform`.
+    """
+
+    if not (160 <= ei <= 1280):
+        print(
+            f"Error: Unsupported EI{ei:d} requested, must be 160 <= EI <= 1280".format(
+                ei
+            )
+        )
+        return False
```

Comment:
Not sure if returning False eventually triggers a throw, but it might be clearer to throw here to make it clear that an inaccurate transform would not be generated.